app.py

```diff
@@ -0,0 +1,274 @@
+"""Blogly application."""
+from flask_debugtoolbar import DebugToolbarExtension
+from flask import Flask, request, redirect, render_template, flash
+from models import db, connect_db, User, Post, Tag
+
+app = Flask(__name__)
+app.config['SQLALCHEMY_DATABASE_URI'] = 'postgresql:///blogly'
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+app.config['SQLALCHEMY_ECHO'] = True
+
+connect_db(app)
+db.create_all()
+
+app.debug = False
+app.config['SECRET_KEY'] = "asdfjkl"
+debug = DebugToolbarExtension(app)
+
+
+@app.route("/")
+def home():
+    posts = Post.query.order_by(Post.created_at.desc()).limit(5).all()
+    return render_template("homepage.html", posts=posts)
+
+
+@app.errorhandler(404)
+def page_not_found(e):
+    """Custom 404 Page"""
+
+    return render_template('404.html'), 404
+
+
+@app.route("/users")
+def list_users():
+    """List users and show add form button"""
+
+    users = User.query.order_by(User.last_name, User.first_name).all()
+
+    return render_template("index.html", users=users)
+
+
+@app.route("/users/new")
+def users_new_form():
+    """Show a form to create a new user"""
+    users = User.query.all()
+    return render_template('usersform.html', users=users)
+
+
+@app.route("/users/new", methods=["POST"])

```

Comment:
The REST `create` route typically is a `POST` on the resource's base path. You might be running into a collision with the route at l.42, and because it only accepts `GET`, `POST` isn't allowed.
```suggestion
@app.route("/users", methods=["POST"])

```